### PR TITLE
Fix `pre-main-kyma-watcher-kcp-tests` && `pre-main-kyma-watcher-kcp-lint` errors

### PR DESCRIPTION
**Pull request by khlifi411**
####  Proposed changes
- Update Makefile to fix failing pipeline
- Fix lint errors

### CODE REVIEW

File: kcp/Makefile [hunk 1]
```diff
@@ -100,31 +96,59 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
...
+client-gen: ## Download client-gen
+	GOBIN=$(LOCALBIN) go install k8s.io/code-generator/cmd/client-gen@v0.23.4
+
+.PHONY: informer-gen
+informer-gen: ## Download informer-gen
+	GOBIN=$(LOCALBIN) go install k8s.io/code-generator/cmd/informer-gen@v0.23.4
+
+.PHONY: lister-gen
+lister-gen: ## Download lister-gen
+	GOBIN=$(LOCALBIN) go install k8s.io/code-generator/cmd/lister-gen@v0.23.4
```
Comment (jakobmoellersap) [inline kcp/Makefile:144]: I think these onese you do not really need. Maybe remove those?

File: kcp/Makefile [hunk 2]
```diff
@@ -100,31 +96,59 @@ deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in
...
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+## Tool Binaries
+KUSTOMIZE ?= $(LOCALBIN)/kustomize
+CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+ENVTEST ?= $(LOCALBIN)/setup-envtest
+CLIENT_GEN ?= $(LOCALBIN)/client-gen
+INFORMER_GEN = $(LOCALBIN)/informer-gen
+LISTER_GEN = $(LOCALBIN)/lister-gen
```
Comment (jakobmoellersap) [inline kcp/Makefile:110]: Also probably need to remove those

**Review by jakobmoellersap — APPROVED** [review]
Great Stuff!

File: kcp/Makefile (hunk 1 above)
Comment (khlifi411) [reply kcp/Makefile:144]: Will be removed [here](https://github.com/kyma-project/kyma-watcher/pull/24/commits/7f26122c9b6ff756ea754a73a7556b52c131d22a)

File: kcp/Makefile (hunk 2 above)
Comment (khlifi411) [reply kcp/Makefile:110]: Will be removed [here](https://github.com/kyma-project/kyma-watcher/pull/24/commits/7f26122c9b6ff756ea754a73a7556b52c131d22a)

**Pull request merged**